h5pt.go

```diff
@@ -41,6 +44,15 @@ func (t *Table) Close() error {
 	}
 	err := h5err(C.H5PTclose(t.id))
 	t.id = 0
+
+	for _, ptr := range t.strings {
```

Comment:
I haven't managed to get a clear spec/promise from the hdf5 docs, but it seems (experiment-programming) that `H5PTappend` doesn't hold onto the memory one passes to it (our `cData` variable in `Append`).

so, the lifetime of `t.strings` and `t.structs` could be restricted to just `Append`:
- (re)allocating them
- freeing them
could all be done inside `Append` and allow for more reasonnable memory usage during long-running hdf5.Table appending.

Reply:
We actually had so many issues with that in the initial development. Therefore, we introduced those pointer slices. BUT: I tried now again - and it just worked fine. :+1: